### PR TITLE
Improve and update ArduinoJson

### PR DESCRIPTION
**Pull request by jputcu**
Specify library name and version:  **ArduinoJson/6.21.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jputcu** [comment]
Building `v6.21.0` within v1 pipeline isn't able to enforce the `C++11` compiler, so that currently gets a `INVALID_CONFIGURATION`.
~~~~
def validate(self):
        if Version(self.version) >= "6.21.0":
            check_min_cppstd(self, 11)
~~~~

File: recipes/arduinojson/all/conanfile.py [hunk 1]
```diff
@@ -17,15 +19,20 @@ class ArduinojsonConan(ConanFile):
...
         self.info.clear()
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def source(self):
+        has_arduinojson_root=Version(self.version) < "6.18.2"
         get(self, **self.conan_data["sources"][self.version],
-            destination=self.source_folder, strip_root=True)
+            destination=self.source_folder, strip_root=has_arduinojson_root)
```
Comment (uilianries) [inline recipes/arduinojson/all/conanfile.py:35]: Could you please explain what error you have when not using stripped root?

Comment (jputcu) [reply recipes/arduinojson/all/conanfile.py:35]: Did a quick local test to force it again, set to `True`. Version `6.18` builds, version `6.21.0` gives the following error:

~~~~
ERROR: arduinojson/6.21.0: Error in source() method, line 34
	get(self, **self.conan_data["sources"][self.version],
	ConanException: The zip file contains more than 1 folder in the root
~~~~

Comment (uilianries) [reply recipes/arduinojson/all/conanfile.py:35]: Thank you for checking it, indeed it extracts to the root directly, so does not need to strip another root folder.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**